### PR TITLE
[MAINT] use pytest-reporter-html1 to create test reports

### PR DESCRIPTION
**Pull request by Remi-Gau**
Closes none

Just trying to see if there are some pytest plugin we can use that help us navigate the gigantic output of the test suite.

Other options:

- https://github.com/Teemu/pytest-sugar
- https://github.com/nicoddemus/pytest-rich

See also options that build full HTML reports:
- https://pytest-html.readthedocs.io/en/latest/index.html
- https://github.com/christiansandberg/pytest-reporter-html1
- https://github.com/prashanth-sams/pytest-html-reporter

### CODE REVIEW

**Comment by Remi-Gau** [comment]
image

OK that's nice but I still need to scroll 100000 lines to just know that.

From a quick look https://github.com/christiansandberg/pytest-reporter-html1 seems to be the most flexible as it uses jinja templates to adapt the reports in any way we want.

**Comment by Remi-Gau** [comment]
@man-shu I think that this may be interesting as an issue for the hackathon if someone is interested.

It'd be a matter of checking what pytest extensions exists: I only did a quick check and see what the reports they generate look like and how to satisfy our needs.

If you think this is a good idea I can draft an issue.

**Comment by man-shu** [comment]
> If you think this is a good idea I can draft an issue

Yes, I agree it would be cool for the hackathon

**Comment by bthirion** [comment]
I'm not  sure about that one: we want to minimize warnings rather than hiding them...

**Comment by Remi-Gau** [comment]
> I'm not sure about that one: we want to minimize warnings rather than hiding them...

I think we agree on that.

One of the issue at the moment is to have an easy way to access what warning is being thrown how many times and by what part of the code base in a full run of the test suite: parsing this from the pytest output at the moment is very tedious.

File: maint_tools/templates/index.html
Comment (Remi-Gau) [inline maint_tools/templates/index.html:1]: taken and tweaked from the upstream repo for the pytest pytest-reporter-html1 extension

Comment (Remi-Gau) [reply maint_tools/templates/index.html:1]: opened an issue to see if the tweaks could be incorporated upstream to facilitate reuse / inheritance in downstream repos

Comment (Remi-Gau) [reply maint_tools/templates/index.html:1]: https://github.com/christiansandberg/pytest-reporter-html1/issues/23

**Comment by Remi-Gau** [comment]
Current look of the report:

Downloadable from the github log
https://github.com/nilearn/nilearn/actions/runs/9661846191/job/26650455610#step:7:27

image

image


image

**Review by bthirion — APPROVED** [review]
The proof is in the pudding. I'm happy with it. Maybe I would wait to hear from other coredevs.

**Comment by Remi-Gau** [comment]
> Maybe I would wait to hear from other coredevs.

yeah this one is not super urgent, also if I can get some changes accepted in the upstream pytest plugin it would make this PR even smaller.

**Comment by Remi-Gau** [comment]
> if I can get some changes accepted in the upstream pytest plugin

that just happened so I will trim down this PR

**Comment by bthirion** [comment]
Still good to me.

**Comment by Remi-Gau** [comment]
Pressed the merge button on the wrong PR. Reverting and reopneing.

**Pull request merged**